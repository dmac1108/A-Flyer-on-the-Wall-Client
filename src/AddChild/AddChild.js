import React, {Component} from 'react'
import FlyersContext from '../FlyersContext'
import './AddChild.css'

class AddChild extends Component {
    state = {
        name: '',
        childrenAdded: [],
    }

    static contextType = FlyersContext
    
    onInputChange = (event) =>{
        const value = event.target.value;
        this.setState({
            [event.target.name]: value,
        })
    }


    handleAddChild = (e) =>{
        e.preventDefault()
        this.context.onAddChild(this.state.name)
        this.setState({
            childrenAdded: [...this.state.childrenAdded, this.state.name]
        })
        console.log(this.state.childrenAdded)
        this.setState({
            name: '',
           
        })
    }
    
    render(){
        const childList = this.state.childrenAdded.map((child)=><li key={child}>{child}</li>)
        return(
            <section className="addchildsection">
                <div>
            <form  className="addChild" onSubmit={(e)=>this.handleAddChild(e)}>
                <legend>Add Child</legend>
                <label htmlFor="add-child">Name</label>
                <input name="name" type="text" id="add-child" onChange={(e)=>this.onInputChange(e)} value={this.state.childName}/>
                <button type="submit">Add Child</button>
                <button type="reset" onClick={()=>this.props.history.push('/flyers')}>Done</button>
            </form>
            </div>
            <div className="addchildlist">
                <h2>Children Added</h2>
            <ul>
            {childList}
            </ul>
            </div>
            </section>
        )
    }
}

export default AddChild
