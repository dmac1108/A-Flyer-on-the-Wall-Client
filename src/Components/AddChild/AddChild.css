.addChild > legend {
    text-align: center;
}
.addchildsection{
    position: relative;
    top: 120px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-start;
}
.addChild {
    
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-content: center;
    width: 65%;
}
.addChild > button {
    width: 50%;
}

.addchildlist{
    
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 25%;
}

.addchildlist > ul >li {
    display: block;
    font-size: 16px;
    margin: 0px;
}

.addchildlist > h2 {
    font-size: 16px;
    text-align: center;
}

@media only screen and (min-width: 550px){
    .addChild{
        width: 40%;
    }
}