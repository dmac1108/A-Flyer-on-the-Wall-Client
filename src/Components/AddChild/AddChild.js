import config from '../../config'
import React, {Component} from 'react'
import FlyersContext from '../../FlyersContext'
import './AddChild.css'
import FlyerApiService from '../../services/flyer-api-service'

class AddChild extends Component {
    state = {
        name: '',
        childrenAdded: [],
    }

    static contextType = FlyersContext
    
    onInputChange = (event) =>{
        const value = event.target.value;
        this.setState({
            [event.target.name]: value,
        })
    }


    handleAddChild = (e) =>{
        e.preventDefault()
        this.context.onAddChild(this.state.name)
        const newChild = {childname: this.state.name}
        FlyerApiService.postChild(newChild)
        .then(()=>{
        this.setState({
            childrenAdded: [...this.state.childrenAdded, this.state.name]
        })
        this.setState({
            name: '',
           
        })
    })
        
        
    }
    
    render(){
        const childList = this.state.childrenAdded.map((child)=><li key={child}>{child}</li>)
        return(
            <section className="addchildsection">
                <div>
            <form  className="addChild" onSubmit={(e)=>this.handleAddChild(e)}>
                <legend>Add Child</legend>
                <label htmlFor="add-child">Name</label>
                <input name="name" type="text" id="add-child" onChange={(e)=>this.onInputChange(e)} value={this.state.childName}/>
                <button type="submit">Add Child</button>
                <button type="reset" onClick={()=>this.props.history.push('/flyers')}>Done</button>
            </form>
            </div>
            <div className="addchildlist">
                <h2>Children Added</h2>
            <ul>
            {childList}
            </ul>
            </div>
            </section>
        )
    }
}

export default AddChild
