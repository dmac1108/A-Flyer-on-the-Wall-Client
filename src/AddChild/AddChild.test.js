import React from 'react'
import AddChild from '././AddChild'
import {BrowserRouter } from 'react-router-dom'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom';




describe('AddChild Component', ()=>{
    it('renders the AddChild Component without crashing', ()=>{
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><AddChild/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the AddChild component', ()=>{
        const wrapper = shallow(<AddChild/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})