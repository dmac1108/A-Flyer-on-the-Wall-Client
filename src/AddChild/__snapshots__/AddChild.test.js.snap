// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddChild Component renders the AddChild component 1`] = `
<section
  className="addchildsection"
>
  <div>
    <form
      className="addChild"
      onSubmit={[Function]}
    >
      <legend>
        Add Child
      </legend>
      <label
        htmlFor="add-child"
      >
        Name
      </label>
      <input
        id="add-child"
        name="name"
        onChange={[Function]}
        type="text"
      />
      <button
        type="submit"
      >
        Add Child
      </button>
      <button
        onClick={[Function]}
        type="reset"
      >
        Done
      </button>
    </form>
  </div>
  <div
    className="addchildlist"
  >
    <h2>
      Children Added
    </h2>
    <ul />
  </div>
</section>
`;
